src/main/java/org/sonar/plugins/stash/client/StashClient.java

```diff
@@ -456,4 +459,33 @@ AsyncHttpClient createHttpClient(String sonarQubeVersion) {
             .build()
     );
   }
+
+  private Collection<JsonObject> performPagedRequest(BoundRequestBuilder requestBuilder,
+      JsonObject body,
+      int expectedStatusCode,
+      String errorMessage) throws StashClientException {
+
+    Collection<JsonObject> result = new ArrayList<>();
+
+    boolean doneLastPage = false;
+    int nextPageStart = 0;
+    // FIXME size/limit support

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1134)